### DISCUSSION TITLE
Sample ray multiple times

### DISCUSSION THREAD

**Discussion by luca-morreale** [mitsuba-renderer/mitsuba3#474] [Q&A] [answered]
Is there any way to cast multiple rays from a camera/sensor in 1 function call? Or at least do it efficiently?

I am referring to `sensor.sample_ray`. When I call that function for several pixels (>1000) the time grows tenfolds.

**Comment by ziyi-zhang** [comment, marked as answer]
Hi,

The argument `Float`/`Point2f` are vectors, so all Mitsuba functions are naturally vectorized.
We can get a vector of `Point2f sample` with `mi.Point2f(np.random.rand(1000, 2))` for debug purpose.
In practice we get them from `sampler.next_2d()` with initialization `sampler.seed(seed, wavefront_size)`. Examples can be found in src/python/python/ad/integrators/common.py

**Comment by luca-morreale** [comment]
That worked, thank you!
